zerver/lib/actions.py

```diff
@@ -2972,6 +2966,25 @@ def bulk_add_subs_to_db_with_logging(
                             if not stream.invite_only]
     return new_occupied_streams
 
+def send_stream_creation_events_for_private_streams(
+    realm: Realm,
+    streams: Iterable[Stream],
+    new_streams: Set[Tuple[int, int]],
```

Comment:
The `new_streams` data structure is awkward, and it should probably be a map from stream ids to list of user_ids.  It would probably avoid the need for `users` to be passed in, and it could simplify the complicated list comprehension below.

Reply:
Yeah, Set[Tuple] is sorta a strange construction; I also feel like its name isn't accurate, since the values in it are tuples where one of the things is a `stream_id`.